week10/bulletin-board/src/pages/Detail.jsx

```diff
@@ -0,0 +1,352 @@
+    } catch (err) {
+      console.error("댓글 삭제 에러:", err);
+    }
+  };
+
+  if (loading) return <div>Loading...</div>;
+  if (error) return <div>Error: {error.message}</div>;
+  if (!post) return <div>No post found</div>;
+
+  return (
+    <PostContainer>
+      <TitleAndHeart>
+        {isEditing ? (
+          <input
+            type="text"
+            value={editedTitle}
+            onChange={(e) => setEditedTitle(e.target.value)}
+            style={{
+              border: "solid 1.5px #ced4da",
+              borderRadius: "5px",
+              height: "2.5vh",
+            }}
+          />
+        ) : (
+          <ContentTitle>{post.title}</ContentTitle>
+        )}
+
+        {isEditing ? (
+          <></>
+        ) : (
+          <Hearts>
+            {" "}
+            <HeartButtonsON onClick={handleAddHeart}>❤ 좋아요</HeartButtonsON>
+            <HeartButtonsOFF onClick={handleRemoveHeart}>
+              💔 좋아요 취소
+            </HeartButtonsOFF>
+          </Hearts>
+        )}
+      </TitleAndHeart>
+      {isEditing ? (
+        <InputText
+          value={editedContent}
+          onChange={(e) => setEditedContent(e.target.value)}
+        />
+      ) : (
+        <p>{post.content}</p>
+      )}
+      {post.image && <img src={post.image} alt={post.title} width="200" />}
+      <p>작성자: {post.nickname}</p>
+      <p>작성일: {new Date(post.createdDate).toLocaleString()}</p>
+      <Likes>{likeCount} 명이 이 글을 좋아합니다. </Likes>
+      <PostButtons>
+        {isEditing ? (
+          <>
+            <FixButton onClick={handleEditPost}>저장</FixButton>
+            <DeleteButton onClick={() => setIsEditing(false)}>
+              취소
+            </DeleteButton>
+          </>
+        ) : (
+          <FixAndDel>
+            <FixButton
+              onClick={() => {
+                setIsEditing(true);
+                setEditedTitle(post.title);
+                setEditedContent(post.content);
+              }}
+            >
+              게시글 수정
+            </FixButton>
+            <DeleteButton onClick={handleDeletePost}>게시글 삭제</DeleteButton>
+          </FixAndDel>
+        )}
+      </PostButtons>
+      <CommentsWrapper>
+        <CommentWrite>
+          <CommentInput
+            value={commentContent}
+            onChange={(e) => setCommentContent(e.target.value)}
+          />
+          <CommentUploadButton onClick={handleAddComment}>
+            댓글 등록
+          </CommentUploadButton>
+        </CommentWrite>
+        <Comments>
+          <CommentBox>
+            {comments.map((comment, index) => (
+              <CommentItem key={index}>
+                <CommentContent>
+                  <CommentAuthor>{comment.commentAuthorNickname}</CommentAuthor>
+                  {comment.content}
+                </CommentContent>
+                <DelIcon
+                  onClick={() =>
+                    handleDeleteComment(comment.commentId || comment.id)
+                  }
+                />
```

Comment:
6. 댓글 삭제 버튼을 렌더링하기 전, 댓글 작성자의 닉네임과 나의 닉네임이 같은지 확인합니다. 같은 경우에만 버튼을 렌더링합니다.

```suggestion
                {myNickname === comment.commentAuthorNickname && (
                  <DelIcon
                    onClick={() =>
                      handleDeleteComment(comment.commentId || comment.id)
                    }
                  />
                )}
```